### PR TITLE
Milestones 0 - Minimal Viable Configuration

### LINKED ISSUE #49: Single "up" and "down" commands to setup/start and tear down environment
Based on the experiences, and as suggested in PR conversations, it is best to have a single command to setup and start the environment. (and tear it down)

See comment https://github.com/ash-sxn/GSoC-2023-docker-based-quickstart/pull/42#pullrequestreview-1492791293

### LINKED ISSUE #51: Create the official working configuration/compose file
move out of the experiment directories.

### PR DESCRIPTION
**Pull request by ash-sxn**
- resolves #49 
- resolves #51 
- created jenkins_init.sh and jenkins_teardown.sh
- updated keygen.sh with absolute path 
- copied files from 02 to the root directory 
- Works locally  on my machine, and on gitpod, didn't check on macOS
- pinging mentors @jmMeessen @berviantoleo @gounthar

### CODE REVIEW

**Review by jmMeessen — COMMENTED** [review]
Tested the PR on my Mac: 
- was puzzled to where to start. Had luck choosing the right file. (=> update README, add validation steps section in PR description) 
- init works, agent is running, demo job is running.
- teardown seems to work
   - had non catastrophic error at the end of the script
      - ``` 
      head -n -1 .tutorials_running.txt
      head: illegal line count -- -1
     ``` 
👍

**Review by jmMeessen — COMMENTED** [review]
tested the GitPod version:
- init: perfect execution
   - jenkins comes up and is correctly configured
   - demo job runs
- teardown: perfect. no error 
 
👍

File: jenkins_terdown.sh
Comment (jmMeessen) [inline jenkins_terdown.sh:1]: there is a typo in the filename. It should be `jenkins_teardown.sh`

**Review by jmMeessen — APPROVED** [review]
LGTM

For a first iteration (Milestone 0) this is a very good implementation. I didn't check the code in details for style but it does the job well. It can be improved and simplified so that people reading it don't get lost and are not mislead by filenames. But this clean up/refactoring is for milestone 3. 
Well done @ash-sxn 👍 


to fix (before merge)
- typo in filename
- update readme to explain how to start.

**Comment by ash-sxn** [comment]
I'll update the Readme in this PR itself 
and I forgot to mention above 
to run the different examples add these arguments to the `./jenkins_init.sh` command 
`1` - 00_old example => `./jenkins_init.sh 1`
`2` - 01_example => `./jenkins_init.sh 2`
`3` - 02_latest_example => `./jenkins_init.sh 3`
`4`- 03 maven_tutorial(doesn't work right now since, that will be covered in next milestone) => `./jenkins_init.sh 4`
If no argument  is used, It runs the latest example present in the root directory docker-compose file

`./jenkins_terdown.sh` is enough for removing any running tutorial (will fix the typo with readme)

**Review by gounthar — APPROVED** [review]
Looks good to me, works fine on GitPod.
Let me try that on my machine too. 👍

**Comment by gounthar** [comment]
It failed. I have to investigate.

```
./jenkins_init.sh 3
+ TUTORIAL=3
+ VAR1=1
+ VAR2=2
+ VAR3=3
+ VAR4=4
+ VARDL=.
+ VAR1L=./experimental_docker_compose_files/00_old_one_from_proposal
+ VAR2L=./experimental_docker_compose_files/01_simple_controller_plus_agent
+ VAR3L=./experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller
+ VAR4L=./experimental_docker_compose_files/03_maven_tutorial
+ DOCKER_COMPOSE='docker compose'
+ check_docker_compose
+ command -v docker
+ docker compose version
+ echo 'Docker Compose is already installed'
Docker Compose is already installed
+ DOCKER_COMPOSE='docker compose'
+ echo docker compose
docker compose
+ [[ 3 == \1 ]]
+ [[ 3 == \2 ]]
+ [[ 3 == \3 ]]
+ generate_ssh_keys ./experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller
+ local tutorial_path=./experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller
+ echo 'generating new ssh keys'
generating new ssh keys
+ sh ./experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller/keygen.sh ./experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller
./experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller/keygen.sh: 35: Syntax error: end of file unexpected (expecting "then")
```

**Comment by ash-sxn** [comment]
@gounthar 
I Found [this](https://stackoverflow.com/questions/18155852/syntax-error-end-of-file-unexpected-expecting-then) 
and added the check_wsl  function in the script please check if it's working now

**Comment by gounthar** [comment]
@ash-sxn : it works for me under WSL2, but I still have to issue:

`find . -name "*sh" -exec dos2unix {} \; -print` before launching the script...

If I don't do that, I get:

```
./jenkins_init.sh 3
-bash: ./jenkins_init.sh: /bin/bash^M: bad interpreter: No such file or directory
```

I'm not so sure we can do anything about it...

**Pull request merged**